src/test/ui/underscore-lifetime/dyn-trait-underscore.stderr

```diff
@@ -1,31 +1,12 @@
-error[E0495]: cannot infer an appropriate lifetime for autoref due to conflicting requirements
-  --> $DIR/dyn-trait-underscore.rs:8:20
-   |
-LL |     Box::new(items.iter())
-   |                    ^^^^
-   |
-note: first, the lifetime cannot outlive the anonymous lifetime #1 defined on the function body at 6:1...
-  --> $DIR/dyn-trait-underscore.rs:6:1
-   |
-LL | / fn a<T>(items: &[T]) -> Box<dyn Iterator<Item=&T>> {
-LL | |     //                      ^^^^^^^^^^^^^^^^^^^^^ bound *here* defaults to `'static`
-LL | |     Box::new(items.iter())
-LL | | }
-   | |_^
-note: ...so that reference does not outlive borrowed content
-  --> $DIR/dyn-trait-underscore.rs:8:14
-   |
-LL |     Box::new(items.iter())
-   |              ^^^^^
-   = note: but, the lifetime must be valid for the static lifetime...
-note: ...so that the expression is assignable
+error[E0621]: explicit lifetime required in the type of `items`
   --> $DIR/dyn-trait-underscore.rs:8:5
    |
+LL | fn a<T>(items: &[T]) -> Box<dyn Iterator<Item=&T>> {
+   |                ---- help: add explicit lifetime `'static` to the type of `items`: `&'static [T]`
```

Comment:
This should be suggesting `-> Box<dyn Iterator<Item=&T> + '_>`, but I'm having trouble getting the correct span. If we manage to improve this error we will also be able to fix the last nitpick I had about the `impl Trait + '_` suggestion which only works for bare `impl Trait` return types (if you had `(impl Trait, )` it wouldn't give the correct suggestion). I thought it would be easy to get the span because it already appeared in the output, but it isn't because it was a comment, not part of the diagnostic. I've seen people be hit by this already in my day job.

Furthermore, I think that all suggestions to add `'static` need to be revised so that we also make it clear that `'static` borrows also include owned values.